src/main/scala/com/gu/zuora/ZuoraService.scala

```diff
@@ -27,24 +30,22 @@ class ZuoraService(config: ZuoraConfig, client: FutureHttpClient, baseUrl: Optio
   def getAccount(accountNumber: String): Future[GetAccountResponse] =
     get[GetAccountResponse](s"accounts/$accountNumber", authHeaders)
 
-  def getAccountIds(identityId: String): Future[List[String]] = {
-    val queryData = QueryData(s"select AccountNumber from account where IdentityId__c = '${identityId.toLong}'")
-    postJson[AccountQueryResponse](s"action/query", queryData.asJson, authHeaders).map(_.records.map(_.AccountNumber))
+  def getAccountFields(identityId: IdentityId): Future[List[DomainAccount]] = {
+    // WARNING constructing queries from strings is inherently dangerous.  Be very careful.
+    val queryData = QueryData(s"select AccountNumber, CreatedSessionId__c from account where IdentityId__c = '${identityId.value}'")
+    postJson[AccountQueryResponse](s"action/query", queryData.asJson, authHeaders).map(_.records.map(DomainAccount.fromWireAccount))
   }
 
-  def getSubscriptions(accountId: String): Future[List[Subscription]] =
-    get[SubscriptionsResponse](s"subscriptions/accounts/$accountId", authHeaders).map(_.subscriptions)
+  def getSubscriptions(accountNumber: ZuoraAccountNumber): Future[List[DomainSubscription]] =
+    get[SubscriptionsResponse](s"subscriptions/accounts/${accountNumber.value}", authHeaders).map { subscriptionsResponse =>
```

Comment:
it's also dangerous to build urls using string concatenation, but admittedly less, as zuora probably won't trust `/../../` in urls, but if it did, and we POSTed something untrusted at the URL level, it might not be ideal.